__test__/pet.test.js

```diff
@@ -0,0 +1,155 @@
+const Pet = require('../src/pet');
+
+describe('constructor', () => {
+    it('returns an object', () => {
+        expect(new Pet('Fido')).toBeInstanceOf(Object);
+    });
+});
+
+describe('constructor', () => {
+    it('sets the name property', () => {
+        const pet = new Pet('Fido');
+        expect(pet.name).toEqual('Fido');
+    });
+});
+
+describe('constructor', () => {
+    it('has an initial age of 0', () => {
+        const pet = new Pet('Fido');
+        expect(pet.age).toEqual(0);
+    });
+});
+
+describe('constructor', () => {
+
+    it('throws an error if the pet is not alive', () => {
+        const pet = new Pet('Fido');
+
+        pet.age = 30;
+
+        expect(() => pet.growUp()).toThrow('Your pet is no longer alive :(');
+    });
+
+    it('increments the age by 1', () => {
+        const pet = new Pet('Fido');
+        pet.growUp();
+        expect(pet.age).toEqual(1);
+    });
+});
+
+describe('constructor', () => {
+    it('increases the hunger property by 5', () => {
+        const pet = new Pet('Fido');
+        pet.growUp();
+        expect(pet.hunger).toEqual(5);
+    });
+});
+
+describe('constructor', () => {
+    it('decreases fitness by 3', () => {
+        const pet = new Pet('Fido');
+        pet.fitness = 10;
+        pet.growUp();
+        expect(pet.fitness).toEqual(7);
+    });
+});
+
+describe('walk', () => {
+
+    it('throws an error is the pet is not alive', () => {
+        const pet = new Pet('Fido');
+
+        pet.fitness = 0;
+
+        expect(() => pet.walk()).toThrow('Your pet is no longer alive :(');
+    });
+
+    it('increases fitness by 4 to a maximum of 10', () => {
+        const pet = new Pet('Fido');
+
+        pet.fitness = 8;
+        pet.walk();
+
+        expect(pet.fitness).toEqual(10);
+    });
```

Comment:
This tests only tests that fitness increases to a maximum of 10, not that it increases by 4. You should have 2 separate tests for each of these scenarios e.g.
```
it('increases fitness by 4', () => {
  const pet = new Pet('Fido');
  pet.fitness = 5;
  pet.walk();

  expect(pet.fitness).toEqual(9);
});

it('increases fitness to a maximum of 10', () => {
  const pet = new Pet('Fido');
  pet.fitness = 8;
  pet.walk();

  expect(pet.fitness).toEqual(10);
});
```